src/goto.jl

```diff
@@ -109,41 +113,54 @@ end
 ## module goto
 
 function GotoItem(mod::Module)
+  name = string(mod)
   file, line = mod == Main ? MAIN_MODULE_LOCATION[] : moduledefinition(mod)
-  GotoItem(string(mod), file, line - 1)
+  GotoItem(name, file, line - 1)
 end
 
 ## toplevel goto
 
-const PathItemsMaps = Dict{String, Vector{ToplevelItem}}
+const PathItemsMap = Dict{String, Vector{GotoItem}}
+
+# NOTE:
+# this data structure should only keep "simple" data so that it doesn't eat
+# too much memory when the cache has got fat (like via `regeneratesymbols`)
+# TODO?:
+# maybe we will end up creating a more generic struct that represents a "symbol",
+# and wanting `SYMBOLSCACHE` to store them instead of `GotoItem`
+# (should be kept "simple" atst though, and ideally it handles both toplevel and local symbols)
+# so that we can use `SYMBOLSCACHE` as a general cache interface for various static features
+# like goto, find references, and so on.
```

Comment:
I believe this would be the better and reusable implementation, but let's leave this as a future ref in this PR.